CHANGELOG.md

```diff
@@ -37,6 +37,77 @@
   // ...
   peopleOver20.unsubscribe();
   ```
+* Added initial support for geospatial queries, with the possibility of querying points. No new data type has been added in this phase, but every embedded object property that conforms to `CanonicalGeoPoint` can be queried.
+The queries can be used to filter object with points that belong to a certain shape following spherical geometry, using the `geoWithin` operator in the `filtered` method. The following shapes are supported in geospatial queries: circle (`GeoCircle` type, defined by its center and radius in radians), box (`GeoBox` type, defined by its bottom left corner and upper right one) and polygon (`GeoPolygon` type, defined by its vertices). Additionally, two new methods have been added, `kmToRadians` and `miToRadians`, that can be used convert kilometers and miles to radians, that are used in the definition of the circle radius. ([#5850](https://github.com/realm/realm-js/pull/5850))  
+Example: 
+```typescript
+class MyGeoPoint implements CanonicalGeoPoint {  //Example of a point class that can be queried with geospatial queries
+  coordinates: GeoPosition;
+  type = "Point" as const;
+
+  constructor(long: number, lat: number) {
+    this.coordinates = [long, lat];
+  }
+
+  static schema: ObjectSchema = {
+    name: "MyGeoPoint",
+    embedded: true,
+    properties: {
+      type: "string",
+      coordinates: "double[]",
+    },
+  };
+}
+
+interface IPointOfInterest {
+  id: number;
+  location: MyGeoPoint;
+}
+
+class PointOfInterest extends Realm.Object implements IPointOfInterest {
+  id = 0;
+  location: MyGeoPoint = new MyGeoPoint(0, 0);
+
+  static schema: ObjectSchema = {
+    name: "PointOfInterest",
+    properties: {
+      id: "int",
+      location: "MyGeoPoint",
+    },
+    primaryKey: "id",
+  };
+}
+
+const berlinCoordinates = new MyGeoPoint(13.397255909303222, 52.51174463251085);
+const radius = kmToRadians(500); //500 kilometers radius, equal to 0.0783932519 radians
+
+const copenhagen: IPointOfInterest = {
+  id: 1,
+  location: new MyGeoPoint(12.558892784045568, 55.66717839648401),
```

Comment:
I agree with your point. I'll keep it like this for the tests because it's easier to work like this, but I'll make `MyGeoPoint` derive from `Realm.Object`, as I think this is what we want to suggest going forward

Reply:
Regarding the example, if I make `MyGeoPoint` derive from `Realm.Object`, would there be an easy way of constructing the object before adding it to Realm. Because like this we need to do something like:

```typescript
    const poi: IPointOfInterest = {
      id: 5,
      location: {
        coordinates: [0, 0],
        type: "Point",
      },
    };
```
What I don't like about this is that one needs to specify `Point`, but I suppose there is no way around it.

Reply:
I'm unsure of which approach we want to use in the examples for the current state of the SDK (`new` or literals). Perhaps @realm/realm-js-team could chime in. But if we do literals, then yeah it'd look like the snippet you just provided.  You could also see if adding `as MyGeoPoint` after the `location` object helps.
```ts
      location: {
        coordinates: [0, 0],
        type: "Point",
      } as MyGeoPoint,
```

Reply:
I just corrected this as we discussed, let's see if the others think some other solution ;)